superset/assets/src/dashboard/components/PropertiesModal.jsx

```diff
@@ -0,0 +1,282 @@
+        value: response.json.pks[i],
+        label: `${user.first_name} ${user.last_name}`,
+      }));
+      this.setState({
+        userOptions: options,
+      });
+    });
+    SupersetClient.get({
+      endpoint: `/api/v1/dashboard/${this.props.dashboardInfo.id}`,
+    }).then(response => {
+      this.setState({ isOwnersLoaded: true });
+      const initialSelectedValues = response.json.result.owners.map(owner => ({
+        value: owner.id,
+        label: owner.username,
+      }));
+      this.onOwnersChange(initialSelectedValues);
+    });
+  }
+
+  onOwnersChange(value) {
+    this.setState(state => ({
+      values: {
+        ...state.values,
+        owners: value,
+      },
+    }));
+  }
+
+  onChange(e) {
+    const { name, value } = e.target;
+    this.setState(state => ({
+      values: {
+        ...state.values,
+        [name]: value,
+      },
+    }));
+  }
+
+  toggleAdvanced() {
+    this.setState(state => ({
+      isAdvancedOpen: !state.isAdvancedOpen,
+    }));
+  }
+
+  save(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    const owners = this.state.values.owners.map(o => o.value);
+    SupersetClient.put({
+      endpoint: `/api/v1/dashboard/${this.props.dashboardInfo.id}`,
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        ...this.state.values,
+        owners,
+      }),
+    })
+      .then(({ json }) => {
+        this.props.addSuccessToast(t('The dashboard has been saved'));
+        this.props.onDashboardSave({
+          title: json.result.dashboard_title,
+          slug: json.result.slug,
+          jsonMetadata: json.result.json_metadata,
+          ownerIds: json.result.owners,
+        });
+        this.props.onHide();
+      })
+      .catch(response =>
+        getClientErrorObject(response).then(({ error, statusText }) => {
+          this.dialog.show({
+            title: 'Error',
+            bsSize: 'medium',
+            bsStyle: 'danger',
+            actions: [Dialog.DefaultAction('Ok', () => {}, 'btn-danger')],
+            body: error || statusText || t('An error has occurred'),
+          });
+        }),
+      );
+  }
+
+  render() {
+    const { userOptions, values, isOwnersLoaded, isAdvancedOpen } = this.state;
+    return (
+      <Modal show={this.props.show} onHide={this.props.onHide} bsSize="lg">
+        <form onSubmit={this.save}>
+          <Modal.Header closeButton>
+            <Modal.Title>
+              <div>
+                <span className="float-left">{t('Dashboard Properties')}</span>
+              </div>
+            </Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Row>
+              <Col md={12}>
+                <h3>Basic Information</h3>
+              </Col>
+            </Row>
+            <Row>
+              <Col md={6}>
+                <label className="control-label" htmlFor="embed-height">
+                  {t('Title')}
```

Comment:
We need to define, if we are going to use backend translation for field label's, descriptions and titles or switch to frontend. By backend translation I mean that the API already has these fields translated to you on the response

Reply:
Do you have docs on how the backend translation works? I'm happy to use that if it works better, but it seems messy to be mixing frontend and backend translations everywhere.

Reply:
My take on this after a short conversation with @nytai and @suddjian is that we generally want to move more things to the frontend and rely less on the framework or framework-related assumptions at this time.

This screen and other CRUD-related screen may evolve to have more non-CRUD related things, and we want for these things to be frontend-centered as opposed to backend-driven. 

I think as we add buttons, actions, headers, descriptions, we want all of the translations to be consistent and on the frontend, as opposed to having a mix backend and frontend declared strings.